tests/unit_tests/steps/test_compile_fortran.py

```diff
@@ -457,3 +458,71 @@ def test_vanilla(self):
                 result = get_mod_hashes(analysed_files=analysed_files, config=config)
 
         assert result == {'foo': 123, 'bar': 456}
+
+
+class Test_get_fortran_preprocessor(object):
+
+    def test_from_env(self):
+        with mock.patch.dict(os.environ, values={'FPP': 'foo_pp --foo'}):
+            fpp, fpp_flags = get_fortran_preprocessor()
+
+        assert fpp == 'foo_pp'
+        assert fpp_flags == ['--foo', '-P']
+
+    def test_empty_env_fpp(self):
+        def mock_run_command(command):
+            if 'fpp' not in command:
+                raise RuntimeError('foo')
+
+        with mock.patch.dict(os.environ, clear=True):
+            with mock.patch('fab.steps.compile_fortran.run_command', side_effect=mock_run_command):
+                fpp, fpp_flags = get_fortran_preprocessor()
+
+        assert fpp == 'fpp'
+        assert fpp_flags == ['-P']
+
+    def test_empty_env_cpp(self):
+        def mock_run_command(command):
+            if 'cpp' not in command:
+                raise RuntimeError('foo')
+
+        with mock.patch.dict(os.environ, values={'FPP': ''}):
```

Comment:
Another case of setting `FPP` to the empty string.

Reply:
:+1: thank you